### DISCUSSION TITLE
Shelly 1PM serial buffer overrun

### DISCUSSION THREAD

**Discussion by host-felice** [arendst/Tasmota#12583] [Q&A] [answered]
Hello, I have a problem with a Shelly 1PM with tasmota: it started to show garbage on the console. When I connect it to the serial to usb adapter the console shows no problem, it happens only when connected to the 220V.
I tried to reset, erase the flash, revert to shelly firmware and again to tasmota with no luck.
I think it is hardware fault, maybe a floating gpio. How could I stop it? Maybe disabling something?
Thanks
![214025942_10223361139430090_1606168434858284464_n](https://user-images.githubusercontent.com/36647269/124639279-45a28a80-de8c-11eb-8c88-a4da20f50c19.jpg)

**Comment by barbudor** [comment, marked as answer]
Indeed it looks like it is receiving wrong data on the serial input
Can you double check your configuration match exactly the template and that there is no serial configuration defined.

If the problem is that the serial RX is floating, you may be able to solve that with a pull-up on the serial connector but if you do so, don't forget to isolate it to avoid any risk of touching.

**Reply by host-felice** [reply]
Thanks for the confirmation! I though so, but since the gpios on the shelly are not isolated from the 220V, I didn't want to "play" with them with resistors. I read the ground is connected to the Neutral, but I didn't read anything about RX/TX. I checked that the GPIO3 is the problematic (if I set it up as switch, the relay starts to go on/off randomly), so I went for the software "solution" setting GPIO3 as Output HI. It is working good, no garbage on the console anymore. Do you think a 4,7k resistor would be a better (but safe) solution?

**Reply by ascillato** [reply]
It seems that you have a unintended soldering between TX and RX pins. Those pins are not used in the Shelly 1PM (https://templates.blakadder.com/shelly_1PM.html).

Just follow their paths to the debug connectors. Please, do that with the device NOT connected to mains.

**Reply by ascillato** [reply]
> setting GPIO3 as Output HI

That is a better solution.

**Reply by host-felice** [reply]
> It seems that you have a unintended soldering between TX and RX pins.

I had not the problem, it started after a while (not sure when since it worked good. I realized it when I checked my circuit breakers and I noticed it blinked the red led without reason). So, is it possible an unintended soldering? And it only shows when connected to the 220V?

**Reply by ascillato** [reply]
> So, is it possible an unintended soldering? And it only shows when connected to the 220V?

Maybe the case is being pressed while you put it in 220V? maybe it has some humidity or dust?

Anyway, with your change in the template, you are not having the crosstalk problem anymore.

**Reply by XeS0r** [reply]
After several hours of debugging this issue was finally solved with the following solution:

**Solution for Shell 1 PM (1st Gen - round design)**:
- Set **GPIO3** in the `Configuration -> Template` to `Output Hi`

Solves:
- `Serial buffer overrun` noise in the console log
